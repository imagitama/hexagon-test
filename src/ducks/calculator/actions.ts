export const CHANGE_INPUT_TEXT = 'CHANGE_INPUT_TEXT'
export const changeInputText = (inputText: string) => ({
  type: CHANGE_INPUT_TEXT,
  payload: {
    inputText
  }
})
