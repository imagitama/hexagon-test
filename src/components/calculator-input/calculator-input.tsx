import React from 'react'
import { connect, MapDispatchToProps } from 'react-redux'
import { bindActionCreators } from 'redux'
import { changeInputText } from '../../ducks/calculator/actions'
import { AppState } from '../../store'
import TextInput from '../text-input/text-input'

interface CalculatorInputProps {
  inputText: string,
  changeInputText: (inputText: string) => void
}

export const CalculatorInput = ({ inputText, changeInputText }: CalculatorInputProps) => (
  <>
    Enter your number:
    <TextInput type="text" defaultValue={inputText} onChange={event => changeInputText(event.target.value)} />
  </>
)

export const mapStateToProps = ({ calculator: { inputText } }: AppState) => ({ inputText })

export const mapDispatchToProps: MapDispatchToProps<{ changeInputText: typeof changeInputText }, {}> = dispatch => bindActionCreators({ changeInputText }, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(CalculatorInput)