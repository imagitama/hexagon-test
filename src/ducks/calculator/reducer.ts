import { Reducer } from 'redux'
import { CHANGE_INPUT_TEXT } from './actions'

interface DefaultState {
  inputText: string
}

const defaultState: DefaultState = {
  inputText: ''
}

const reducer: Reducer<DefaultState> = (state = defaultState, action = { type: '' }) => {
  switch (action.type) {
    case CHANGE_INPUT_TEXT:
      return {
        ...state,
        inputText: action.payload.inputText
      }

    default:
      return state
  }
}

export default reducer
