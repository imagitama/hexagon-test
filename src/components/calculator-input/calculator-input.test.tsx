import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import { CalculatorInput, mapStateToProps, mapDispatchToProps } from './calculator-input'
import { changeInputText } from '../../ducks/calculator/actions'

describe('Calculator input', () => {
  describe('When mounted', () => {
    let component: ShallowWrapper

    beforeAll(() => {
      component = shallow(<CalculatorInput inputText="Hello world!" changeInputText={() => null} />)
    })

    it('Renders correctly', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    describe('When called with mock state', () => {
      it('Maps to props correctly', () => {
        expect(mapStateToProps({
          calculator: {
            inputText: 'Hello world!'
          }
        })).toEqual({
          inputText: 'Hello world!'
        })
      })
    })
  })

  describe('mapDispatchToProps', () => {
    describe('When called with mock actions', () => {
      it('Maps to props correctly', () => {
        const mockDispatch = jest.fn()
        mapDispatchToProps(mockDispatch).changeInputText('Hello world!')

        expect(mockDispatch).toHaveBeenCalledWith(changeInputText('Hello world!'))
      })
    })
  })
})
