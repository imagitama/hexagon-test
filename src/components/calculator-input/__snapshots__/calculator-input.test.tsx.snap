// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator input When mounted Renders correctly 1`] = `
<Fragment>
  Enter your number:
  <Styled(input)
    defaultValue="Hello world!"
    onChange={[Function]}
    type="text"
  />
</Fragment>
`;
